File: backend/src/main/kotlin/com/example/backend/models/Malade.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "malade")
class Malade {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "nom", length = 32)
  var nom: String? = null

  @Column(name = "telephone", nullable = false, length = 32)
  var telephone: String? = null

  @Column(name = "modeReglement", nullable = false, length = 32)
  var modeReglement: String? = null

  @Column(name = "poid", nullable = false)
  var poid: Double? = 0.0

  @Column(name = "taille", nullable = false)
  var taille: Double? = 0.0

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "CodePostal_id")
  var codePostal: CodePostal? = null

  @Column(name = "reduction", nullable = false, length = 32)
  var reduction: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer", nullable = false)
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/malade.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class Malade(Base):
  __tablename__ = "malade"

  id = Column(Integer, primary_key=True)
  nom = Column(String(32))
  telephone = Column(String(32), nullable=False)
  mode_reglement = Column(String(32), nullable=False)
  poid = Column(Float, default=0.0)
  taille = Column(Float, default=0.0)
  code_postal_id = Column(Integer, ForeignKey("code_postal.id"))
  code_postal = relationship("CodePostal", lazy="select")
  reduction = Column(String(32), nullable=False)
  supprimer = Column(Integer, default=0)
